From 3887753a5260c39a0428ca82759992164f44e40e Mon Sep 17 00:00:00 2001
From: "Matias E. Fernandez" <mfernandez@pisco.ch>
Date: Wed, 9 Nov 2011 20:16:39 +0100
Subject: [PATCH] Issue #1336880: fix XPath handling in getIDsFromXML

---
 plugins/sources/xml.inc |   23 ++++++-----------------
 1 files changed, 6 insertions(+), 17 deletions(-)

diff --git a/plugins/sources/xml.inc b/plugins/sources/xml.inc
index 197688d..1c46221 100644
--- a/plugins/sources/xml.inc
+++ b/plugins/sources/xml.inc
@@ -437,18 +437,16 @@ class MigrateItemsXML extends MigrateItems {
 
     $this->cache_ids = NULL;
     $ids = array();
-    $full_xpath = $this->itemXpath . '/' . $this->itemIDXpath;
+    $result = $xml->xpath($this->itemXpath);
 
-    $result = $xml->xpath($full_xpath);
+    $ids = array();
     if ($result) {
-      if (count($result) > 1) {
-        foreach ($result as $id) {
+      foreach ($result as $element) {
+        $id = $this->getItemID($element);
+        if (!is_null($id)) {
           $ids[] = (string)$id;
         }
       }
-      else {
-        $ids[] = (string)$result[0];
-      }
     }
     $this->cache_ids = array_unique($ids);
     return $this->cache_ids;
@@ -507,16 +505,7 @@ class MigrateItemsXML extends MigrateItems {
     $result = $xml->xpath($this->itemXpath);
 
     if ($result) {
-      if (count($result) > 1) {
-        foreach ($result as $item_xml) {
-          $id = $this->getItemID($item_xml);
-          $item = new stdclass;
-          $item->xml = $item_xml;
-          $items[$id] = $item;
-        }
-      }
-      else {
-        $item_xml = $result[0];
+      foreach ($result as $item_xml) {
         $id = $this->getItemID($item_xml);
         $item = new stdclass;
         $item->xml = $item_xml;
-- 
1.7.7.2

